### PR TITLE
Tanvigupta27: Assignment 3

### PR DESCRIPTION
**Pull request by tanvigupta27**
**Language Used:** Python
**Evidence:**
A3_Q1
A3_Q1
A3_Q2
A3_Q2
A3_Q3
A3_Q3
A3_Q4
A3_Q4

### CODE REVIEW

**Comment by RoyalEagle73** [comment]
Update title accordingly.

File: tanvigupta27/A3_Q1.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    print(A)
+    res = 0
+    for i in A:
+        for j in A:
+            if (i%j > res):
+                res = i%j
+    print("Output:",res)
+    return
+maxmod(A)
+maxmod(B)
```
Comment (RoyalEagle73) [inline tanvigupta27/A3_Q1.py]: Add a new line at EOF.

File: tanvigupta27/A3_Q3.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    print("A:",A,"B:",B)
+    while(A!=B):
+        if(A > B):
+            A -= B
+        else:
+            B -= A
+    print("GCD:",A%(10^9))
+    return
+gcd(A,B)
+gcd(C,D)
```
Comment (RoyalEagle73) [inline tanvigupta27/A3_Q3.py]: Add a new line at EOF.

File: tanvigupta27/A3_Q4.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+                    break
+            if(flag == 0) and (i!= 1):
+                print(i)
+    else:
+        print("Exit: N > 300")
+    return
+# if (N>=1 and N <=300):
+prime_no(5)
+prime_no(10)
+prime_no(350)
```
Comment (RoyalEagle73) [inline tanvigupta27/A3_Q4.py]: Add a new line at EOF.

File: tanvigupta27/A3_Q2.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    for i in res:
+        result.extend(i) if isinstance(i, list) else result.append(i)
+        ans = 0
+        for i in range(len(result)):
+            ans = ans^result[i]
+    print(ans)
+    return 
+xor_of_list(A)
+xor_of_list(B)
+xor_of_list(C)
```
Comment (RoyalEagle73) [inline tanvigupta27/A3_Q2.py:21]: Upload O(n) solution please.

Comment (tanvigupta27) [reply tanvigupta27/A3_Q2.py:21]: Added in 558ed61

File: tanvigupta27/A3_Q4.py (hunk 3 above)
Comment (tanvigupta27) [reply tanvigupta27/A3_Q4.py]: Added in 558ed61

File: tanvigupta27/A3_Q3.py (hunk 2 above)
Comment (tanvigupta27) [reply tanvigupta27/A3_Q3.py]: Added in 558ed61

File: tanvigupta27/A3_Q1.py (hunk 1 above)
Comment (tanvigupta27) [reply tanvigupta27/A3_Q1.py]: Added in 558ed61

**Review by RoyalEagle73 — APPROVED** [review]

**Pull request merged**